### PR TITLE
Docs/developer guide intro improvements

### PR DESCRIPTION
**Pull request by tkfu**
This is a grab-bag of edits and fixes for the developer guide. In some cases, it's to correct inaccuracies, in others just clarifications.

### CODE REVIEW

File: docs/ota-client-guide/modules/ROOT/pages/_partials/recommended-steps.adoc [hunk 1]
```diff
@@ -88,14 +83,23 @@ This step is not that different from the equivalent step during the integration
...
+* One role is for signing software metadata, marking it as valid for distribution.
+* One role acts as the root of trust for the main software repository.
+* One role acts as the root of trust for telling vehicles exactly what software they should install, and when.
++
+For each role, you only need to get the key out of storage when the role has to act. That means the first key is only used when new software is released to production, and the two root-of-trust keys are only used in case of a breach, or to revoke a signing key that may have been lost or stolen.
 
 * *Integrate libaktualizr with the client software on your board*
 +
-Once you understand how to the libaktualizr demo application works, you'll want to build a fully-functional integration with your board's software.
+Once you understand how to the libaktualizr demo application works, you'll want to build a fully-functional integration with your board's software and the other ECUs in the vehicle, if desired. Some examples of things you can do:
```
Comment (patrickvacek) [inline docs/ota-client-guide/modules/ROOT/pages/_partials/recommended-steps.adoc]:
```suggestion
Once you understand how the libaktualizr demo application works, you'll want to build a fully-functional integration with your board's software and the other ECUs in the vehicle, if desired. Some examples of things you can do:
```

File: docs/ota-client-guide/modules/ROOT/pages/developer-tools.adoc [hunk 2]
```diff
@@ -16,49 +16,49 @@ You can use it to easily evaluate OTA Connect functionality. By default, it's in
...
 
-You'll be using the common Yocto command `bitbake` to build disk images. You can start with our demo recipes then make your own recipes.
+You'll be using the common Yocto command `bitbake` to build disk images. You can start with our demo recipes, then make your own recipes.
 
 [NOTE]
 ====
 __Related Tool__: *meta-updater*
 
-The "meta-updater" defines some of the important "ingredients" that go into our Yocto recipes. Yocto uses the concept of *layers* to bundle different types of "ingredients". In the case of OTA Connect, our "meta-updater" layer adds the aktualizr client and the OSTree tool to your Yocto builds.
+The "meta-updater" layer defines some of the important "ingredients" that go into our Yocto recipes. Yocto uses the concept of *layers* to bundle different types of "ingredients". In the case of OTA Connect, our "meta-updater" layer adds the aktualizr client and the OSTree tool to your Yocto builds.
```
Comment (patrickvacek) [inline docs/ota-client-guide/modules/ROOT/pages/developer-tools.adoc]: The quotes around "meta-updater" feel strange to me. That is the name, it's not a euphemism or nickname or something.

**Comment by patrickvacek** [comment]
FYI CI is failing just because of timeouts. If you push an update, it'll automatically retry and probably resolve itself.

**Comment by merltron** [comment]
> Cool. Just some minor suggestions.
> 
> What's the advantage of `xref` over `link`?

Essentially, because this: https://docs.antora.org/antora/2.0/navigation/link-syntax-and-content/#page

File: docs/ota-client-guide/modules/ROOT/pages/_partials/recommended-steps.adoc [hunk 3]
```diff
@@ -46,40 +44,37 @@ In a proper production workflow, you'll need separate organizations to manage th
...
+For the integration phase, it's fine if you keep these keys on your development computer. Later, you'll want to keep these keys somewhere more secure.
 
-* *Try the libaktualizr demo app and build recipe*
+* *Start customizing the update client*
 +
-When you move to production, you'll want to integrate OTA functionality into your board's native software. This is more efficient than sending commands to our standalone aktualizr client.
+Most production use cases will have unique requirements that aren't met by the stock aktualizr app. With libaktualizr, the C++ library used to build aktualizr, you can customize every step of the installation process.
 +
-You can use our libaktualizr C++ library to integration all same functionality that standalone aktualizr provides. To prove you with some inspiration, we've created a demo app that integrates libaktualizr. We've also included a build recipe. You can use it to build a disk image that contains the demo app.
+To prove you with some inspiration, we've created a very simple demo app that integrates libaktualizr, along with a Yocto layer to build it into your test board.
```
[the remark this answers is not in the document]
Comment (merltron) [reply docs/ota-client-guide/modules/ROOT/pages/_partials/recommended-steps.adoc]: > And is it worth providing the actual links?
Yes probably, I started doing links for "evaluate" then stopped there.

Also concepts like "layer" kind need explaining somewhere - probably on *our* Yocto page.?Thats why I avoided using the domain-specific terminology initially.

File: docs/ota-client-guide/modules/ROOT/pages/developer-tools.adoc (hunk 2 above)
Comment (tkfu) [reply docs/ota-client-guide/modules/ROOT/pages/developer-tools.adoc]: Yeah, I think I agree. I'll push a quick fix.

**Comment by tkfu** [comment]
> What's the advantage of `xref` over `link`?

[Special syntax in antora.](https://docs.antora.org/antora/2.0/asciidoc/page-to-page-xref/#xref-and-page-id-anatomy)

File: docs/ota-client-guide/modules/ROOT/pages/_partials/recommended-steps.adoc [hunk 4]
```diff
@@ -46,40 +44,37 @@ In a proper production workflow, you'll need separate organizations to manage th
...
+Although it's convenient to use shared-credential provisioning so that the OTA Connect server manages the whole process, using shared-credential provisioning means that a private key gets transmitted from the OTA Connect server to each vehicle. For automotive-grade security, most large fleet owners will want to use a TPM or other type of hardware security module, so that the vehicle's private key material never has to exist outside of the vehicle.
 +
-To install your own device certificates, you need a fleet root certificate and private key to sign them. Installation can be tricky, so we've provided some instructions on how to generate your own self-sign certificate and simulate device-credential provisioning for testing.
+In the initial integration phase you might not have the TPM/HSM ready to use, but you can still start using device-credential provisioning, using a self-signed certificate and loading signed device certificates onto your test devices using our tooling and instructions.
```
[the remark this answers is not in the document]
Comment (merltron) [reply docs/ota-client-guide/modules/ROOT/pages/_partials/recommended-steps.adoc]: > might not have the TPM/HSM ready to use

@tkfu  Sorry I dont understand how this relates to whether you have to use a self-signed certificate or not. I thought it was just to save the work of getting a proper cert regardless of hardware (whats a TPM by the way?)

**Comment by merltron** [comment]
> > What's the advantage of `xref` over `link`?
> 
> [Special syntax in antora.](https://docs.antora.org/antora/2.0/asciidoc/page-to-page-xref/#xref-and-page-id-anatomy)

I just realized they probably wont work if you click 'em while reading on GitHub or the local source dir. Maybe we really need to be more aggressive in encouraging people to read online.

File: docs/ota-client-guide/modules/ROOT/pages/_partials/recommended-steps.adoc (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrickvacek) [reply docs/ota-client-guide/modules/ROOT/pages/_partials/recommended-steps.adoc]: It's true, "layer" is not clear with an explanation for a newbie. I am hesitant to replicate Yocto docs, but a brief explanation might be in order as well as a link to the proper source of truth.

**Comment by patrickvacek** [comment]
> I just realized they probably wont work if you click 'em while reading on GitHub or the local source dir. Maybe we really need to be more aggressive in encouraging people to read online.

I'm having that realization as well the more I look at this and play around with it. I'd prefer to still be able to present the information via github without resorting to the portal, but I think we should encourage users to go there unless they have a good reason not to.

**Comment by tkfu** [comment]
The xrefs will work in most cases--they'll only break if you're linking to content in a different repo. And since the only other repo we'd be linking to for now is on our private gitlab, anyway, it doesn't really matter.

**Comment by merltron** [comment]
> I'm having that realization as well the more I look at this and play around with it. I'd prefer to still be able to present the information via github without resorting to the portal, but I think we should encourage users to go there unless they have a good reason not to.

We can get around the issue with conditional statements (if env = github) but we then need to come up with a schema for defining xrefs.

**Comment by merltron** [comment]
> The xrefs will work in most cases--they'll only break if you're linking to content in a different repo. And since the only other repo we'd be linking to for now is on our private gitlab, anyway, it doesn't really matter.

Well, I'm working on incorporating meta-updater content right now - they'll be a few more cross repo xrefs, but I see if I can get around this issue.

**Comment by tkfu** [comment]
The solution, I think, would be to add the following to the top of each file:

```
ifndef::env-github[:ota-client-guide: ota-client-guide::]
ifdef::env-github[:ota-client-guide: /advancedtelematic/aktualizr/blob/master/docs/ota-client-guide/modules/ROOT/pages/]
```

Then, when you need to xref across modules, you'd slightly modify the syntax:

```
xref:{ota-client-guide}filename.adoc[link text]
```

**Comment by patrickvacek** [comment]
(Don't know what's up with the DCO but don't let that stop you from merging if you're ready to.)

**Pull request merged**